wgsl/index.bs

```diff
@@ -426,59 +426,104 @@ The result type depends on the number of letters provided. Assuming a `vec4<f32>
 
 ## Texture Types ## {#texture-types}
 
-* type must be `f32`, `i32` or `u32`
-* The parameterized type for the images is the type after conversion from reading/sampling. E.g. you can have an image with texels with 8bit unorm components, but when you sample them you get a 32-bit float result (or vec-of-f32).
+* The parameterized type for the images is the type after conversion from reading/sampling.
+    E.g. you can have an image with texels with 8bit unorm components, but when you sample
+    them you get a 32-bit float result (or vec-of-f32).
 
 ### Sampled Texture ### {#texture-sampled}
 
 <pre class='def'>
 `texture_sampled_1d<type>`
-  %1 = OpTypeImage %type 1 0 0 0 1 Unknown
+  %1 = OpTypeImage %type 1D 0 0 0 1 Unknown
+
+`texture_sampled_1d_array<type>`
+  %1 = OpTypeImage %type 1D 0 1 0 1 Unknown
 
 `texture_sampled_2d<type>`
-  %1 = OpTypeImage %type 2 0 0 0 1 Unknown
+  %1 = OpTypeImage %type 2D 0 0 0 1 Unknown
+
+`texture_sampled_2d_array<type>`
+  %1 = OpTypeImage %type 2D 0 1 0 1 Unknown
 
 `texture_sampled_3d<type>`
-  %1 = OpTypeImage %type 3 0 0 0 1 Unknown
+  %1 = OpTypeImage %type 3D 0 0 0 1 Unknown
 
 `texture_sampled_2d_ms<type>`
-  %1 = OpTypeImage %type 2 0 0 1 1 Unknown
+  %1 = OpTypeImage %type 2D 0 0 1 1 Unknown
+
+`texture_sampled_2d_ms_array<type>`
+  %1 = OpTypeImage %type 2D 0 1 1 1 Unknown
+
+`texture_sampled_cube<type>`
+  %1 = OpTypeImage %type Cube 0 0 0 1 Unknown
+
+`texture_sampled_cube_array<type>`
+  %1 = OpTypeImage %type Cube 0 1 0 1 Unknown
 </pre>
+* type must be `f32`, `i32` or `u32`
+
 
 ### Read-only Storage Texture ### {#texture-ro}
 <pre class='def'>
-`texture_ro_1d<type>`
-  %1 = OpTypeImage %type 1 0 0 0 2 Unknown
+`texture_ro_1d<image_storage_type>`
```

Comment:
should the parameter just be called `format` (for consistency with API' `GPUTextureFormat`)?

Reply:
Not sure, `format` isn't very descriptive. This name matches the name in the grammar.